### PR TITLE
[pt2] test if core decomps are differentiable

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107241

### CODE REVIEW

File: test/test_decomp.py [hunk 1]
```diff
@@ -44,12 +44,19 @@ def overload_to_aten_name(overload):
...
     for op in op_db
     if op.aten_name in decomposition_names
     or op.aten_backward_name in decomposition_names
 ]
+_decomp_test_ops_core_autograd = [
+    op
+    for op in op_db
+    if op.aten_name in core_decomposition_names
+    and op.supports_autograd
+]
```
Comment (nkaretnikov) [inline test/test_decomp.py:59]: I can also filter out anything containing substring "backward" here, should I?

File: test/test_decomp.py [hunk 2]
```diff
@@ -403,6 +410,40 @@ def test_unsupported(t):
     return any(test_unsupported(x) for x in itertools.chain(flat_args, flat_kwargs))
 
 
+core_backward_failures = {
```
Comment (nkaretnikov) [inline test/test_decomp.py:413]: Some of these take way too long, so I ran pytest with the timeout argument and skipped those.

**Review by ezyang — APPROVED** [review]

File: test/test_decomp.py (hunk 2 above)
Comment (jithunnair-amd) [reply test/test_decomp.py:413]: Hi @nkaretnikov, can you please share the timeout setting and command you used? We need to run them on ROCm to see which tests take way too long on ROCm and pptentially add skips for those.

Comment (nkaretnikov) [reply test/test_decomp.py:413]: @jithunnair-amd I think it was this: `time pytest --timeout=360 -rfxs test/test_decomp.py -k test_quick_core_backward -v 2>&1 | tee decomp.log`. You need to install https://pypi.org/project/pytest-timeout/ for the timeout option, which is why I only ran it locally and not used a decorator. This is not a dependency that PyTorch has.

**Pull request closed without merging**